README.md

```diff
@@ -1,5 +1,345 @@
+`?_storyblok=<timestamp>`) to the preview URL, so it should be okay to a use a
+constant URL base (rather than the always changing SHA-based Vercel preview
+URLs).
+
+There is a fair amount of code in the codebase that integrates with Storyblok.
+For example, there is a bridge that syncs the site with changes coming from the
+Storyblok editor.
+
+There is also some middleware that checks if the HTTP referrer is
+`https://app.storyblok.com/`. In that case, the code assumes that the end user
+is coming to the site from within the Storyblok iframe. So it puts the user in
+Next.js preview mode. When the user is in preview mode, the website shows a
+yellow banner at the top of the page, letting the user know that they are seeing
+both published and unpublished content on the site.
+
+### Vercel
+
+Each website corresponds to a separate project in Vercel, quansight-consulting
+and quansight-labs, respectively. However, because both projects are mapped to
+the same repository on GitHub, whenever a commit is made to the repo, whether
+the commit was just for the Consulting site or just for the Labs site, it
+triggers a preview deployment for both sites. Likewise, whenever a merge is made
+to the `main` branch, it triggers a production deployment to both websites.
+
+Each Vercel project has settings that allow it to integrate with Storyblok,
+Next.js, and GitHub. Each Vercel project has three separate environments:
+development, preview and production. The development and preview environments
+contains the preview key to Storyblok. The production environment contains the
+public key to Storyblok.
+
+Each Vercel project is also configured with a webhook that Storyblok uses to
+kick off a preview deploy of the `develop` branch whenever content is published
+in Storyblok.
+
+### Slack
+
+There is a Vercel app for Slack. It is set up to send deployment notifications
+to the `#vercel-notifications` channel.
+
+### Next.js plus the codebase in this repo
+
+The code in this repo is built on top of the Next.js framework. Next.js has a
+feature called preview mode. When preview mode is turned on (via a cookie), the
+Next.js framework dynamically renders each page; otherwise it serves a static,
+pre-rendered page (except when doing local development and then it always
+dynamically renders each page).
+
+The code in this repo takes advantage of the Next.js preview mode to integrate
+better with Storyblok and Vercel. When the code detects preview mode, it passes
+`{"version": "draft"}` to the Storyblok API in order to preview content that has
+been saved but not yet published.
+
+There is also code that detects when the request comes from a Storyblok iframe.
+In this case, it automatically enters into Next.js preview mode.
+
+To help the end user distinguish what view of the site they are seeing, the
+codebase defines a visual banner at the top of each page. When the site is in
+preview mode, the banner turns green and displays a message telling the user
```

Comment:
Yeah, something with less of a good/bad status-quality affect might be good.

Purple for Published, and gray for Saved, maybe?

Reply:
I like that suggestion, but could we reverse it? Purple for saved, gray for published?

Reply:
I'm ok either way, I could argue either side of the semantics of "what should be marked with the more prominent color, Saved or Published?"

Reply:
My reasoning:

- Gray is more muted, doesn't call so much attention to itself, which I think is suggestive of the idea that published is closer to what you'll see on production, so maybe it doesn't need to call attention to itself. Whereas when you're looking at saved content, you're seeing different content than what you'll see in the next deploy, so maybe we should draw attention to that fact with an attention grabbing color.
- I know my brain (and maybe other people's too) is going to think of Next.js preview mode as being turned on/off. I think gray suggests off, and when paired against gray, the color purple suggests on.

Reply:
On your second bullet, I would actually argue the other way. The primary stored-data version you interact with in the Storyblok editor is Saved, not Published -- so for that, gray = Saved would represent 'no difference from what you see in Storyblok' whereas purple = Published represents 'this may be different from what you see in Storyblok'

Reply:
Yeah, it really depends on one's perspective. 

But I wonder... if we use [saved=purple, published=gray] then when you're making changes in Storyblok you'll see the purple banner in the preview iframe. So then later, if you look at a vercel.app preview URL that has been popped out of Next.js preview mode (so now it's showing a gray banner and published content), would the ABSENCE of a vivid color like purple from the banner that you're used to seeing be enough to catch your attention?

Reply:
Mmm, great point.

Yeah, you've convinced me, i like your way better.

Just struck me, though: we should probably pick a color that *clashes* with the site colors, so not purple. Bright white? Orange?